File: resources-management/src/main/resources/static/js/service/fileService.js
app.service("fileService",function ($http) {
    this.listFiles = function () {
        return $http.get("/manage/api/file/list");
    }

    this.createDirectory = function (obj) {
        return $http.post("/manage/api/file/create/directory",obj);
    }

    this.createDocument = function (form) {
        return $http({
            url: "/manage/api/file/create/document",
            method: "post",
            transformRequest: angular.identity,
            headers:{
                'Content-Type': undefined
            },
            data: form
        })
    }

    this.deleteFile = function (obj) {
        return $http.post("/manage/api/file/delete",obj);
    }

    this.rename = function (obj) {
        return $http.post("/manage/api/file/rename",obj);
    }
})

File: resources-management/rm-common/src/main/java/com/mfs/rmcommon/service/FileService.java
package com.mfs.rmcommon.service;

import com.mfs.rmcommon.po.Request;
import com.mfs.rmcommon.po.Response;

import java.io.IOException;

public interface FileService {

    Response createDirectory(Request request);

    Response createDocument(Request request);

    Response deleteFile(Request request);

    Response renameFile(Request request);

    Response listFiles(Request request);
}
